peerconnection.go

```diff
@@ -949,18 +968,18 @@ func (pc *PeerConnection) openSRTP() {
 				return
 			}
 
-			receiver.Track.Kind = codec.Type
-			receiver.Track.Codec = codec
-			pc.newRTPTransceiver(
-				receiver,
-				nil,
-				RTPTransceiverDirectionRecvonly,
-			)
+			receiver.Track().mu.Lock()
+			receiver.Track().kind = codec.Type
+			receiver.Track().codec = codec
+			receiver.Track().mu.Unlock()
 
-			pc.onTrack(receiver.Track)
+			if pc.onTrackHandler != nil {
+				pc.onTrack(receiver.Track(), receiver)
+			} else {
+				pcLog.Warnf("OnTrack unset, unable to handle incoming media streams")
```

Comment:
Would it make more sense to check this right away at the top of the function and return an error if it's not set? Seems like it might reduce user confusion if they forget to set an OnTrack handler.